sqlalchemy_exasol/base.py

```diff
@@ -1199,3 +1279,33 @@ def fkey_rec():
     def get_indexes(self, connection, table_name, schema=None, **kw):
         """EXASolution has no explicit indexes"""
         return []
+
+    def type_descriptor(self, typeobj):
+        """Return a DB-specific TypeEngine for a generic SA type.
+
+        We wrap DateTime columns so their Python values serialize cleanly for pyexasol.
+        """
+        if isinstance(typeobj, sqltypes.DateTime):
+            return EXATimestamp()
+        return super().type_descriptor(typeobj)
+    
+    # leave this for true DB-API remapping (ODBC etc.)
+    dbapi_exception_translation_map = {}
+
+    def do_execute(self, cursor, statement, parameters, context=None):
+        # print("TRIGGERED DO EXECUTE")
+        try:
+            return super().do_execute(cursor, statement, parameters, context)
+        except Exception as e:
+            mapped = _PYEXA_TO_SA.get(e.__class__.__name__)
```

Comment:
We might be able to simplify the exceptions here, as the Python Exception case allows for more than 1 to be checked, and we could extract that to a function if other usages needed to be covered.

However, it's possible you have something more in mind, as it looks like `do_execute` maybe doesn't cover all execution points.